lib/storage/data/external/AwsClient.js

```diff
@@ -140,78 +211,107 @@ class AwsClient {
-                'from datastore', err, this._dataStoreName);
-                return callback(retError);
+        })).then(data => callback(null, data))
+            .catch(err => {
+            let logLevel;
+            let retError;
+            if (err instanceof NotFound) {
+                logLevel = 'info';
+                retError = errors.LocationNotFound;
+            }
+            else {
+                logLevel = 'error';
+                retError = errorInstances.ServiceUnavailable.customizeDescription(
+                    `Error returned from ${this.type}: ${err.message}`);
             }
-            return callback(null, data);
+            logHelper(log, logLevel, 'error heading object ' +
+                'from datastore', err, this._dataStoreName);
+            return callback(retError);
         });
     }
     get(objectGetInfo, range, reqUids, callback) {
         const log = createLogger(reqUids);
         const { key, dataStoreVersionId } = objectGetInfo;
-        const request = this._client.getObject({
+        const params = {
             Bucket: this._awsBucketName,
             Key: key,
             VersionId: dataStoreVersionId,
-            Range: range ? `bytes=${range[0]}-${range[1]}` : null,
-        }).on('success', response => {
-            log.trace(`${this.type} GET request response headers`,
-                { responseHeaders: response.httpResponse.headers,
-                    backendType: this.clientType });
-        });
-        const stream = request.createReadStream();
-
-        // modify the stream destroy() behavior to also abort the HTTP request
-        const streamDestroy = stream._destroy.bind(stream);
-        stream._destroy = (err, cb) => {
-            log.debug('aborting GET request in progress', { objectGetInfo });
-            request.abort();
-            if (streamDestroy) {
-                streamDestroy(err, cb);
-            }
+            Range: range ? `bytes=${range[0]}-${range[1]}` : undefined,
         };
-        stream.on('error', err => {
-            let logLevel;
-            if (err.code === 'NotFound') {
-                logLevel = 'info';
-            } else {
-                logLevel = 'error';
+        this._client.send(new GetObjectCommand(params)).then(data => {
+            let stream = data.Body;
+            // ChecksumStream is a Duplex but doesn't properly expose .pipe()
+            // Always use the underlying source stream for reliability
+            // AWS SDK v3 returns streams that may not be fully compatible Node.js Readable streams
+            // They might have .pipe() but not .on(), .once(), etc.
+            // Always check if we have a proper Node.js stream with full API
+            const isProperNodeStream = stream && 
+                typeof stream.pipe === 'function' && 
+                typeof stream.on === 'function' &&
+                typeof stream.once === 'function';
+            
+            if (!isProperNodeStream) {
```

Comment:
AWS SDK (v3) will not return a "proper" node.js stream anymore, so no point trying to handle that
case. The only case we need to handle is AWS SDK v3 stream.

Reply:
c.f. previous comment, change needed already : we don't want to introduce dead code